src/pages/NewSignup.vue

```diff
@@ -144,22 +160,46 @@ export default {
     },
   },
   methods: {
-    getOptions(field) {
-      if (field.dependant) {
-        if (this.userData[field.dependantField])
-          return field.dependantFieldMapping[
-            this.userData[field.dependantField]
-          ];
-      } else return field.options;
+    showBasedOn() {
+      return Object.keys(this.formData.attributes).forEach((field) => {
+        let fieldAttributes = this.formData.attributes[field];
+        let showBasedOn = fieldAttributes.showBasedOn;
```

Comment:
is it fine to have method and variable with same name

Reply:
it's a local variable. Scope is not beyond the function. I don't see a problem.